### ISSUE TITLE
[maple] kernel warning: "don't recognize rcg frequency for ahb_clk_src"

### ISSUE THREAD

**Issue by jamuir** [sonyxperiadev/bug_tracker#58]
**Platform:** yoshino
**Device:** maple
**Kernel version:** 4.4
**Android version:** 7.1.1

**Description**
When the kernel (revision `337e9f2`) boots it issues the following warning:

```
[    1.125250] msm-thermal soc:qcom,msm-thermal: probe_psm:Failed reading node=/soc/qcom,msm-thermal, key=qcom,pmic-sw-mode-temp. err=-22. KTM continues
[    1.125773] msm_thermal:vdd_restriction_reg_init Defer regulator vdd-gfx probe
[    1.126044] msm_thermal:probe_vdd_rstr Err regulator init. err:-517. KTM continues.
[    1.126341] msm-thermal soc:qcom,msm-thermal: probe_vdd_rstr:Failed reading node=/soc/qcom,msm-thermal, key=qcom,max-freq-level. err=-517. KTM continues
[    1.127612] msm_mpm_dev_probe(): Cannot get clk resource for XO: -517
[    1.129592] i2c-msm-v2 c179000.i2c: probing driver i2c-msm-v2
[    1.129866] i2c-msm-v2 c179000.i2c: error on clk_get(core_clk):-517
[    1.130147] i2c-msm-v2 c179000.i2c: error probe() failed with err:-517
[    1.130995] i2c-msm-v2 c17a000.i2c: probing driver i2c-msm-v2
[    1.131384] i2c-msm-v2 c17a000.i2c: error on clk_get(core_clk):-517
[    1.131567] i2c-msm-v2 c17a000.i2c: error probe() failed with err:-517
[    1.133334] i2c-msm-v2 c1b5000.i2c: probing driver i2c-msm-v2
[    1.133631] i2c-msm-v2 c1b5000.i2c: error on clk_get(core_clk):-517
[    1.133908] i2c-msm-v2 c1b5000.i2c: error probe() failed with err:-517
[    1.135385] i2c-msm-v2 c1b6000.i2c: probing driver i2c-msm-v2
[    1.135794] i2c-msm-v2 c1b6000.i2c: error on clk_get(core_clk):-517
[    1.135950] i2c-msm-v2 c1b6000.i2c: error probe() failed with err:-517
[    1.157001] branch_clk_handoff: gcc_mss_mnoc_bimc_axi_clk clock is enabled in HW even though ENABLE_BIT is not set
[    1.158140] qcom,gcc-8998 100000.qcom,gcc: Registered GCC clocks
[    1.161160] don't recognize rcg frequency for ahb_clk_src
[    1.161384] ------------[ cut here ]------------
[    1.161530] WARNING: at ffffff8008b836e8 [verbose debug info unavailable]
[    1.161781] 
[    1.161927] CPU: 4 PID: 6 Comm: kworker/u16:0 Not tainted 4.4.21-g2585df3-02468-g1e480b5-dirty #6
[    1.162178] Hardware name: SoMC Maple-ROW(MSM8998 v2.1) (DT)
[    1.162332] Workqueue: deferwq deferred_probe_work_func
[    1.162715] task: ffffffc0f94a4100 ti: ffffffc0f94d8000 task.ti: ffffffc0f94d8000
[    1.162975] PC is at _rcg_clk_get_parent+0x198/0x1dc
[    1.163117] LR is at _rcg_clk_get_parent+0x198/0x1dc
[    1.163255] pc : [<ffffff8008b836e8>] lr : [<ffffff8008b836e8>] pstate: 60000145
[    1.163503] sp : ffffffc0f94dbaa0
[    1.163642] x29: ffffffc0f94dbaa0 x28: ffffffc0f94cc000 
[    1.164152] x27: ffffffc0f947c800 x26: 00000000000000a8 
[    1.164551] x25: ffffff80095a7c28 x24: ffffff800937e238 
[    1.165059] x23: ffffff800937e218 x22: 00000000000000a8 
[    1.165455] x21: ffffff80095b4538 x20: 0000000000000000 
[    1.165969] x19: ffffff80095b4518 x18: 0000000013ab6680 
[    1.166364] x17: 000000000bebc200 x16: ffffff80095a6cf0 
[    1.166873] x15: 00000000152bec00 x14: 0ffffffffffffffe 
[    1.167272] x13: 0000000000000018 x12: 0101010101010101 
[    1.167784] x11: ffffffc0f94db6f0 x10: ffffff8009635000 
[    1.168189] x9 : 0000000000000000 x8 : 00de00f700de0087 
[    1.168701] x7 : 00fe00e500540096 x6 : 0000000000000000 
[    1.169103] x5 : ffffffc0f8b01280 x4 : 0000000000000000 
[    1.169500] x3 : 0000000000000000 x2 : ffffffc0f94d8000 
[    1.170008] x1 : 0000000000000000 x0 : 000000000000002d 
[    1.170409] 
[    1.170409] PC: 0xffffff8008b836a8:
[    1.170662] 36a8  6b08001f 54fffd41 b9402060 6b06001f 54fffce1 17fffff1 b0005041 f9400662
[    1.172098] 36c8  91074021 aa1303e0 91008021 d63f0040 f9401a61 f0003fc0 9111e000 97d6d79f
[    1.173414] 36e8  d4210000 d2800000 17ffffe6 f9401a61 f0003fc0 91116000 97d6d798 d4210000
[    1.174823] 3708  d2800000 17ffffdf f9401801 f0003fc0 9110c000 97d6d791 d4210000 d2800000
[    1.176239] 
[    1.176239] LR: 0xffffff8008b836a8:
[    1.176378] 36a8  6b08001f 54fffd41 b9402060 6b06001f 54fffce1 17fffff1 b0005041 f9400662
[    1.177784] 36c8  91074021 aa1303e0 91008021 d63f0040 f9401a61 f0003fc0 9111e000 97d6d79f
[    1.179187] 36e8  d4210000 d2800000 17ffffe6 f9401a61 f0003fc0 91116000 97d6d798 d4210000
[    1.180490] 3708  d2800000 17ffffdf f9401801 f0003fc0 9110c000 97d6d791 d4210000 d2800000
[    1.181906] 
[    1.181906] SP: 0xffffffc0f94dba60:
[    1.182157] ba60  08b836e8 ffffff80 f94dbaa0 ffffffc0 08b836e8 ffffff80 60000145 00000000
[    1.183459] ba80  00000513 00000000 00000513 00000000 ffffffff ffffffff 00000000 00000000
[    1.184871] baa0  f94dbac0 ffffffc0 08b83744 ffffff80 00000014 00000000 08b80c98 ffffff80
[    1.186283] bac0  f94dbad0 ffffffc0 08b80bbc ffffff80 f94dbb00 ffffffc0 08b80ec4 ffffff80
[    1.187587] 
[    1.187860] ---[ end trace ae02c4fd112e6981 ]---
[    1.187998] Call trace:
[    1.188137] Exception stack(0xffffffc0f94db8d0 to 0xffffffc0f94dba00)
[    1.188392] b8c0:                                   ffffff80095b4518 0000008000000000
[    1.188537] b8e0: ffffffc0f94dbaa0 ffffff8008b836e8 ffffff8009207338 ffffff8009490ee8
[    1.188796] b900: 0000000100000000 ffffff8009636198 ffffffc0f94db920 ffffff80080e9ed8
[    1.189054] b920: ffffffc0f94db9c0 ffffff80080ea244 ffffff80080ea20c 0000000000000000
[    1.189309] b940: ffffff80095b4538 00000000000000a8 ffffff800937e218 ffffff800937e238
[    1.189453] b960: ffffff80095a7c28 00000000000000a8 000000000000002d 0000000000000000
[    1.189708] b980: ffffffc0f94d8000 0000000000000000 0000000000000000 ffffffc0f8b01280
[    1.189965] b9a0: 0000000000000000 00fe00e500540096 00de00f700de0087 0000000000000000
[    1.190109] b9c0: ffffff8009635000 ffffffc0f94db6f0 0101010101010101 0000000000000018
[    1.190364] b9e0: 0ffffffffffffffe 00000000152bec00 ffffff80095a6cf0 000000000bebc200
[    1.190618] [<ffffff8008b836e8>] _rcg_clk_get_parent+0x198/0x1dc
[    1.190761] [<ffffff8008b83744>] rcg_clk_get_parent+0x18/0x20
[    1.191020] [<ffffff8008b80bbc>] __handoff_clk.part.11+0x38/0x314
[    1.191161] [<ffffff8008b80ec4>] __handoff_clk+0x2c/0x5c
[    1.191415] [<ffffff8008b81758>] msm_clock_register+0x130/0x2c0
[    1.191557] [<ffffff8008b81960>] of_msm_clock_register+0x78/0xa0
[    1.191813] [<ffffff8008b8ce30>] msm_mmsscc_8998_probe+0x2b8/0x39c
[    1.191960] [<ffffff800859292c>] platform_drv_probe+0x40/0xc4
[    1.192217] [<ffffff8008591314>] driver_probe_device+0x1c0/0x2e4
[    1.192359] [<ffffff800859155c>] __device_attach_driver+0x84/0xb0
[    1.192615] [<ffffff800858f8d4>] bus_for_each_drv+0x54/0x94
[    1.192758] [<ffffff8008591054>] __device_attach+0xc0/0x128
[    1.193014] [<ffffff80085915c0>] device_initial_probe+0x10/0x18
[    1.193155] [<ffffff800858fb84>] bus_probe_device+0x90/0x98
[    1.193410] [<ffffff8008590a54>] deferred_probe_work_func+0x78/0xac
[    1.193553] [<ffffff80080b36ac>] process_one_work+0x13c/0x324
[    1.193807] [<ffffff80080b39d0>] worker_thread+0x13c/0x4b4
[    1.193952] [<ffffff80080b9894>] kthread+0xd0/0xe4
[    1.194208] [<ffffff80080845f0>] ret_from_fork+0x10/0x20
```

**Symptoms**

**How to reproduce**
Capture kernel logs shortly after boot:  `adb shell dmesg > dmesg.out`

**Comment by jerpelea** [comment]
@kholk

**Comment by jerpelea** [comment]
@kholk

**Comment by jerpelea** [comment]
the bug does not appear on 6.4 branch

**Comment by jerpelea** [comment]
@jamuir can you please check 6.4 and back port the fix to 5.7

**Comment by jerpelea** [comment]
I can not replicate the bug on 5.7 (Android 7.1)
